controller_manager:
  ros__parameters:
    update_rate: 1000 # Hz
    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster
    velocity_controller:
      type: velocity_controllers/JointGroupVelocityController
    position_controller:
      type: position_controllers/JointGroupPositionController

velocity_controller:
  ros__parameters:
    joints:
      - main_wheel_joint # 主驱动轮
position_controller:
  ros__parameters:
    joints:
      - fork_joint #叉齿
      - main_wheel_base_joint #舵轮轴
    gains:
      main_wheel_base_joint:
        p: 60
        d: 1
        i: 4
        i_clamp: 1.0
      fork_joint:
        p: 30
        d: 0
        i: 0
        i_clamp: 1.0
